tests/Aspire.Hosting.Tests/Health/ResourceHealthCheckServiceTests.cs

```diff
@@ -129,12 +130,13 @@ public async Task HealthCheckIntervalDoesNotSlowBeforeSteadyHealthyState()
     {
         using var builder = TestDistributedApplicationBuilder.Create(testOutputHelper);
 
-        AutoResetEvent? are = null;
+        var are = new AutoResetEvent(false);
```

Comment:
nit: Instead of AutoResetEvent, what about using a Channel? AddCheck can write to the channel, and the test body can read from it. Avoids blocking the test thread.